### PR TITLE
fix(exporter): separate distribution from repo file structure

### PR DESCRIPTION
**Pull request by decko**
## Summary
- `parse_content_path()` was including RPM repo-internal paths (`Packages/<letter>/`, `repodata/`) in the distribution name
- Added `_strip_repo_structure()` to detect and strip these suffixes before building the distribution string
- Returns `None` when stripping leaves an empty distribution (defensive against malformed paths)
- Added end-to-end distribution assertions to integration tests

**Example (from PULP-1889):**
- Path: `/api/pulp-content/public-copr/pbrezina/s4u/fedora-44-x86_64/Packages/p/python3-libipa_hbac-2.14.0-1.fc44.x86_64.rpm`
- Before: distribution = `pbrezina/s4u/fedora-44-x86_64/Packages/p`
- After: distribution = `pbrezina/s4u/fedora-44-x86_64`

Closes PULP-1889

## Test plan
- [x] All 37 tests pass
- [x] RPM `Packages/<letter>/` stripping verified
- [x] RPM `repodata/` stripping verified
- [x] RPM paths without repo structure unchanged
- [x] Empty distribution after stripping returns `None`
- [x] Integration tests assert distribution field end-to-end (Python + RPM)
- [x] Ruff format clean

Assisted-by: Claude Opus 4.6 (1M context) <noreply@anthropic.com>

## Summary by Sourcery

Adjust content path parsing to exclude RPM repository-internal directories from distributions and strengthen tests around parsed distribution values.

Bug Fixes:
- Stop including RPM repo-internal paths like Packages/<letter> and repodata in the parsed distribution field.
- Return None when stripping repository structure would result in an empty distribution, avoiding malformed parsed records.

Enhancements:
- Refine content path parsing helpers to separate repository structure from logical distribution segments.
- Improve code style and slicing semantics in content parsing utilities for better readability.

Tests:
- Extend unit and integration tests to cover RPM Packages directory stripping, repodata stripping, unchanged RPM paths without repo structure, empty distributions, and distribution propagation into parquet output for both Python and RPM.

### CODE REVIEW

File: management_tools/pulp-access-logs-exporter/tests/test_content_export.py [hunk 1]
```diff
@@ -165,15 +169,36 @@ def test_python_path(self):
...
+            "/api/pulp-content/ccac33ac/templates/kernel-core-6.8.0-300.fc40.x86_64.rpm"
+        )
+        assert result is not None
+        assert result["domain"] == "ccac33ac"
+        assert result["distribution"] == "templates"
+        assert result["filename"] == "kernel-core-6.8.0-300.fc40.x86_64.rpm"
+
+    def test_empty_distribution_after_stripping_returns_none(self):
+        result = parse_content_path("/api/pulp-content/domain/repodata/repomd.xml")
+        assert result is None
```
[the remark this answers is not in the document]
Comment (decko) [reply management_tools/pulp-access-logs-exporter/tests/test_content_export.py:200]: Added in 5b50aa1 — `test_skips_empty_distribution_after_stripping` verifies `convert_content_to_arrow_table` drops the record end-to-end when `parse_content_path` returns `None` due to empty distribution.

**Comment by decko** [comment]
Re: @sourcery-ai general review comments —

**Relaxing `RPM_PACKAGES_DIR_RE` to `[a-zA-Z0-9]+`**: The single-char constraint is intentional. RPM repos using `nested_alphabetically` layout (which is what COPR and all our production repos use) always have a single-letter bucket under `Packages/`. Relaxing to `+` would risk false positives — a legitimate distribution segment like `Packages/some-tool` would be incorrectly stripped. The `flat` layout (`Packages/<filename>.rpm` without a letter subdir) is a known limitation we'll validate against prod logs and handle in a follow-up if needed.

**Guarding `_strip_repo_structure` to RPM context**: The patterns (`Packages/<single-char>` and `repodata`) are specific enough to be unambiguous in practice — no Pulp distribution would realistically be named "repodata" or have a `Packages/b` suffix. Making it content-type-conditional would add complexity without meaningful safety gain, since the function already returns early on the first match and `parse_content_path` returns `None` for the edge case where stripping leaves an empty distribution.

**End-to-end test**: Added in 5b50aa1 per the inline suggestion.
